File: 2D Simulation/MATLAB/SOLVER.m
function [B]= SOLVER(N,A,B,IP)
%	REAL A(N,N),B(N),T
%	INTEGER IP(N)
%	SOLUTION OF LINEAR SYSTEM, A*X=B.
%	N=ORDER OF MATRIX.
%	B=RIGHT HAND SIDE VECTOR.
%	IP=PIVOT VEVTOR OBTAINED FROM SUBROUTINE DECOMP.
%	B=SOLUTION VECTOR X.
%
	if N~=1
        NM1=N-1;
        for K=1:NM1
            KP1=K+1;
            M=IP(K);
            T=B(M);
            B(M)=B(K);
            B(K)=T;
            for I=KP1:N
                B(I)=B(I)+A(I,K)*T;
            end
        end
        for KB=1:NM1
            KM1=N-KB;
            K=KM1+1;
            B(K)=B(K)/A(K,K);
            T=-B(K);
            for I=1:KM1
                B(I)=B(I)+A(I,K)*T;
            end
        end
    end
	B(1)=B(1)/A(1,1);

end

File: 2D Simulation/Original Conversion/src/SOLVER.py
import numpy as np


def SOLVER(N, A, B, IP):
    """
    Real A(N, N), B(N), T
    Integer IP(N)
    Solution of Linear System A*X = B
    N = Order of Matrix
    B = Right Hand Side Vector
    IP = Pivot vector obtained from subroutine decomp.
    B = Solution vector x
    """

    if N != 1:
        NM1 = N - 1
        for K in range(1, NM1 + 1):
            KP1 = K + 1
            M = IP[K - 1]
            T = B[M - 1]
            B[M - 1] = B[K - 1]
            B[K - 1] = T
            for I in range(KP1, N + 1):
                B[I - 1] = B[I - 1] + A[I - 1, K - 1] * T

        for KB in range(1, NM1 + 1):
            KM1 = N - KB
            K = KM1 + 1
            B[K - 1] = B[K - 1] / A[K - 1, K - 1]
            T = -B[K - 1]

            for I in range(1, KM1 + 1):
                B[I - 1] = B[I - 1] + A[I - 1, K - 1] * T

    B[0] = B[0] / A[0, 0]
    return B
